src/modules/users/users.service.ts

```diff
@@ -437,22 +438,27 @@ export class UsersService {
     }
   }
 
-  async softDeleteUser(id: string): Promise<void> {
+  async softDeleteUser(idToDelete: string, userId: string): Promise<void> {
     try {
-      const user = await this.getUserById(id);
+      const userToDelete = await this.getUserById(idToDelete);
+      const user = await this.getUserById(userId);
```

Comment:
but we are sure that user exist, we check it in the validation guard, no?

Reply:
But below I don't check if the user exists, I need this to check if the userToDelete has the same ID as user or has the admin role to delete user. In my opinion I should have these lines of code

Reply:
why line 444 needed? you should already have it from the guard